rules/Php81/NodeFactory/EnumFactory.php

```diff
@@ -91,14 +98,70 @@ private function createEnumCaseFromConst(ClassConst $classConst): EnumCase
         return $enumCase;
     }
 
-    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode): EnumCase
+    /**
+     * @param array<int|string, mixed> $mapping
+     */
+    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode, array $mapping = []): EnumCase
     {
         /** @var MethodTagValueNode $nodeValue */
         $nodeValue = $phpDocTagNode->value;
-
+        $enumValue = $mapping[$nodeValue->methodName] ?? $nodeValue->methodName;
         $enumName = strtoupper($nodeValue->methodName);
-        $enumExpr = $this->builderFactory->val($nodeValue->methodName);
+        $enumExpr = $this->builderFactory->val($enumValue);
 
         return new EnumCase($enumName, $enumExpr);
     }
+
+    /**
+     * @return array<int|string, mixed>
+     */
+    private function generateMappingFromClass(Class_ $class): array
+    {
+        $mapping = [];
+
+        $classMethod = $class->getMethod('values');
+        if ($classMethod === null || !is_array($classMethod->stmts)) {
+            return $mapping;
+        }
+
+        foreach ($classMethod->stmts as $methodStmt) {
+            if (
+                !($methodStmt instanceof Return_)
+                || !($methodStmt->expr instanceof Expr\Array_)
+            ) {
+                continue;
+            }
```

Comment:
What if there are `Return_` instanceof `If_`? I think you can use `BetterNodeFinder->findInstancesOfInFunctionLikeScoped()`

```php
$returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($classMethod, Return_::class);
```

and then loop the returns

Reply:
I see, the `if` condition possibly never happen on enum class `values` method :)

Reply:
Nice one.